app/concepts/lockstep/client.rb

```diff
@@ -115,6 +122,7 @@ def request(method, path, body, params)
       request["SdkType"] = 'Ruby'
       request["SdkVersion"] = '2022.4.32.0'
       request["MachineName"] = Socket.gethostname
+      request["LS-InternalService"] = internal_service_key if magic_link_call?
```

Comment:
This should be more generic. It shouldn't specifically check for `magic_link_call`. We should check if `internal_service_key` is set. If it is set, we need to set the value.